client/src/components/events-components/EventCard.tsx

```diff
@@ -7,11 +7,12 @@ import { deleteEventRsvp, eventRsvp } from "@/utils/eventService";
 
 interface EventProps {
   event: EventWithRsvp;
-  editable: boolean;
+  eventEditable: boolean;
 }
 
-const EventCard = ({ event, editable }: EventProps) => {
+const EventCard = ({ event, eventEditable }: EventProps) => {
   const [isEditing, setIsEditing] = useState(false);
+  const [displayedEvent, setDisplayedEvent] = useState(event);
   const [isRsvpByCurrentUser, setIsRsvpByCurrentUser] = useState(
     event.isRsvpCurrentUser
```

Comment:
I would consider moving all these states out and keep `EventCard` pure presentational component